### DISCUSSION TITLE
Is there a way to have all scripts and css in files and not inline?

### DISCUSSION THREAD

**Discussion by darkrain-nl** [jpanther/congo#68] [Ideas]
Hi all,

Love this theme, works great.

I'm trying to not have to use unsafe_inline in my CSP headers, this is almost working, but it would be a bit easier if the theme didn't create inline script (and css) tags.

Would this be easy to accomplish, I couldn'd find a quick way, so hence this discussion instead of a PR...

What do you think?

**Comment by jpanther** [comment]
Thanks for pointing this out. I'll admit this is not something I had considered when implementing some features. I'm always interested in privacy and security so this is something I'm happy to look at in more detail. I presume the biggest culprit here is the dark mode switching and some of the vendor scripts? Perhaps I should look at moving these items to their own JS files instead. Out of interest, where are you seeing inline CSS?

**Reply by darkrain-nl** [reply]
I didn't see any inline CSS, I just mentioned that because I'm not sure if I use all the options the theme has...

It was indeed the dark mode switching that made me create this discussion. That always has a script inline, even when forcing dark mode ...

Thanks for looking into this.
